lib/Jsrt/ChakraCore.h

```diff
@@ -163,4 +163,346 @@ JsGetModuleHostInfo(
     _In_ JsModuleHostInfoKind moduleHostInfo,
     _Outptr_result_maybenull_ void** hostInfo);
 
+/// <summary>
+///     Called by the runtime to load the source code of the serialized script.
+///     The caller must keep the script buffer valid until the JsSerializedScriptUnloadCallback.
+///     This callback is only supported by the Win32 version of the API
+/// </summary>
+/// <param name="sourceContext">The context passed to Js[Parse|Run]SerializedScriptWithCallback</param>
+/// <param name="script">The script returned.</param>
+/// <returns>
+///     true if the operation succeeded, false otherwise.
+/// </returns>
+typedef bool (CHAKRA_CALLBACK * JsSerializedLoadScriptCallback)
+    (JsSourceContext sourceContext, JsValueRef *value,
+    JsParseScriptAttributes *parseAttributes);
+
+/// <summary>
+///     Create String Variable from C String
+/// </summary>
+/// <remarks>
+///     <para>
+///     Experimental. We may update the name or behavior until it is stable.
+///     </para>
+/// </remarks>
+/// <param name="content">Pointer to string memory.</param>
+/// <param name="length">Length of the string.</param>
+/// <param name="value">.</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsCreateString(
+        const char *content,
+        size_t length,
+        _Out_ JsValueRef *value);
+
+/// <summary>
+///     Create String Variable from Utf8 String
+/// </summary>
+/// <remarks>
+///     <para>
+///     Experimental. We may update the name or behavior until it is stable.
+///     </para>
+/// </remarks>
+/// <param name="content">Pointer to string memory.</param>
+/// <param name="length">Length of the string.</param>
+/// <param name="value">.</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsCreateStringUtf8(
+        const uint8_t *content,
+        size_t length,
+        _Out_ JsValueRef *value);
+
+/// <summary>
+///     Create String Variable from Utf16 String
+/// </summary>
+/// <remarks>
+///     <para>
+///     Experimental. We may update the name or behavior until it is stable.
+///     </para>
+/// </remarks>
+/// <param name="content">Pointer to string memory.</param>
+/// <param name="length">Length of the string.</param>
+/// <param name="value">.</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsCreateStringUtf16(
+        const uint16_t *content,
+        size_t length,
+        _Out_ JsValueRef *value);
+
+/// <summary>
+///     Write string value into C string buffer
```

Comment:
>Write string value into C string buffer [](start = 8, length = 39)

Please describe more details of this API. What if start/length out of range. What if buffer is null. What assumption we have on buffer size.